src/core/net/ip6.cpp

```diff
@@ -1121,7 +1121,8 @@ Error Ip6::HandleDatagram(OwnedPtr<Message> aMessagePtr, const void *aLinkMessag
         }
 #endif
 
-        forwardHost = header.GetDestination().IsMulticastLargerThanRealmLocal();
+        // Forward multicast packets to host network stack unless they came from there.
+        forwardHost = !aMessagePtr->IsOriginHostUntrusted();
```

Comment:
With this we pass all received MLE traffic to the host (all link-local and mesh-local traffic received by device over radio would be forwarded to host)

I understand it may be what is expected in some cases, but can it cause harm in any situation?

Reply:
Good point, we probably should filter out MLE and TMF multicast. I'll update the PR.

Reply:
@abtink It turns out we already filtered out MLE and TMF in PassToHost();

Reply:
closing this comment, feel free to re-open if I missed anything. @abtink